### ISSUE TITLE
Generate better coverage data

### ISSUE THREAD

**Issue by migurski** [openaddresses/openaddresses-ops#14]
Companies like Google and HERE produce coverage summaries of their data:

* https://developers.google.com/maps/coverage
* https://developer.here.com/rest-apis/documentation/geocoder/topics/coverage-geocoder.html

We’ve got a visual version of this in the maps at https://results.openaddresses.io, but a tabular form might be useful as well.

**Comment by iandees** [comment]
Mapbox: https://www.mapbox.com/geocoding/#coverage

**Comment by migurski** [comment]
That Mapbox map is in dire need of visual design help.

**Comment by migurski** [comment]
Making some progress on this [in `openaddresses/population` repo](https://github.com/openaddresses/population), starting with per-country area and population estimates and writing to a useful database that can drive a coverage page:

![EU](http://mike.teczno.com/img/oa-coverages/coverage-EU.png)

![NA](http://mike.teczno.com/img/oa-coverages/coverage-NA.png)

**Comment by iandees** [comment]
I <3 those colors.

**Comment by nvkelso** [comment]
Cool! Reminds me about those funky Natural Earth concordances I need to fix.

**Comment by migurski** [comment]
I have a temporary representation of this data here: http://ec2-54-89-13-7.compute-1.amazonaws.com

Feedback at this stage would be really helpful!

**Comment by nvkelso** [comment]
Nice! I like the addition of the population map at the bottom.

I continue to think the break from substantial and minimal should be
somewhere in the 10% to 20% range.

**Comment by nvkelso** [comment]
The text in the intro helps explain the 3 parts of the table, but if you breeze over that it's not clear why the table parts are grouped that way. Might also add a column for the overall coverage grade (full, substantial, minimal)? This would allow future sorting of the table, too.

**Comment by migurski** [comment]
Good ideas, I incorporated them all.

**Comment by iandees** [comment]
This looks great, thanks for putting it together! The columns are a tiny bit confusing, though. maybe the "population in land area" column should be "population covered" or "estimated population covered"?

**Comment by migurski** [comment]
Okay, made that change. I experimented with splitting up the columns a bit, curious if this makes sense.

Old:

<img width="971" alt="screen shot 2017-03-14 at 2 26 39 pm" src="https://cloud.githubusercontent.com/assets/58730/23923124/953b9720-08c2-11e7-8b4d-c7a2c223d6d6.png">

New:

<img width="974" alt="screen shot 2017-03-14 at 2 26 05 pm" src="https://cloud.githubusercontent.com/assets/58730/23923127/9901e54e-08c2-11e7-88e9-071e9a526b0a.png">

**Comment by nvkelso** [comment]
Is there special HTML formatting that would allow these to copy into OpenOffice as a table? Right now I'd need to do processing to get it into a full table (as Land Area Covered, Population Covered, and Address Density all have sub cells that copy into 1 generic OpenOffice cell):

<img width="715" alt="screen shot 2017-03-14 at 14 49 54" src="https://cloud.githubusercontent.com/assets/853051/23923835/829c76d6-08c5-11e7-91ed-7c5d8e2c79a4.png">

**Comment by migurski** [comment]
I was thinking of this with the "new" variant from the comment above, with separate columns for the parenthesized parts. I'll push it live; tell me if it works for you.

**Comment by migurski** [comment]
Results when pasting into Numbers:

<img width="752" alt="screen shot 2017-03-14 at 3 07 56 pm" src="https://cloud.githubusercontent.com/assets/58730/23924465/109ba6a8-08c8-11e7-9b7b-04db8ea51918.png">

**Comment by nvkelso** [comment]
YES!

<img width="944" alt="screen shot 2017-03-14 at 15 16 34" src="https://cloud.githubusercontent.com/assets/853051/23924753/438bec84-08c9-11e7-9f30-43b59947a7d3.png">

**Comment by migurski** [comment]
✨ Spreadsheets! ✨

**Comment by nvkelso** [comment]
Now add a save as DBF option ;) JK!

**Comment by nvkelso** [comment]
The paragraphs at the top serve as legend, but are missing two items:

- Coverage Group – move this out of the intro text.
- Population Coverage – which summarize and link people to an new anchor tag below the table for full info. 

<img width="453" alt="legend-markup" src="https://cloud.githubusercontent.com/assets/853051/23927525/81b8a8d6-08d7-11e7-9504-77c2a199c458.png">

**Comment by migurski** [comment]
After some good brainstorming with @sbma44 and Michael Steffen, I’m going to dumb down the address density number a little. It’s currently the _average of densities_ (with standard deviation) and I’m going to simplify it to the _average density_ so the arithmetic is easier to understand.

**Comment by michaelsteffen** [comment]
@migurski you can call me by my github name, it's OK :).

👋 @iandees @nvkelso

**Comment by NelsonMinar** [comment]
Loving this work. My requests:
1. Can't wait for a map view like the preview you teased up above! Maybe color by coverage group?
2. To facilitate scanning, please right align all numbers. (Or align on the decimal point, but a trailing zero is probably easier.)
3. Also for scanning, don't mix thousands and millions. Ie instead of 563K and 17M people make it 563k and 17000k. (or 0.56M and 17.00M)
4. The Palantino font's use of descending numerals and inconsistent numeral heights makes this really hard. Maybe this is worse in Windows, see [screenshot attached](https://cloud.githubusercontent.com/assets/382136/23961626/bf1428cc-0968-11e7-92d1-1445a3422d68.PNG)

**Comment by migurski** [comment]
Good suggestions, @NelsonMinar. I’ve applied them all except the font thing. Can you tell what font is actually being rendered on Windows? It’s not Palatino, which looks like this for me:

<img width="374" alt="screen shot 2017-03-15 at 10 26 49 am" src="https://cloud.githubusercontent.com/assets/58730/23961999/f09e03a8-0969-11e7-92c6-76061b16c6cc.png">

**Comment by NelsonMinar** [comment]
You're right, I think it's actually Georgia that my sample is showing (second option in the CSS for body). My Windows 10 system doesn't have a font named "Palatino", but it does have "Palatino Linotype" which seems to work in the CSS and [looks OK](https://cloud.githubusercontent.com/assets/382136/23962463/686778be-096b-11e7-897b-f22be98f17f6.PNG).

**Comment by migurski** [comment]
This is real now: http://results.openaddresses.io/coverage

**Comment by nvkelso** [comment]
Congrats, great new feature :)

**Comment by iandees** [comment]
👏  This is really excellent. I tweeted about it here: https://twitter.com/openaddr/status/844666719209750528

Can we add a map like the green/red one on https://results.openaddresses.io/ but coloring countries based on population coverage percentage?

**Comment by migurski** [comment]
Yes, absolutely! I probably shouldn't use the same colors for this.

**Comment by jharpster** [comment]
These coverage metrics are the best thing ever. Also wondering if raw counts of addresses by country are maintained some place or if that would be useful to anyone besides me.

**Comment by migurski** [comment]
That sounds pretty useful. Maybe a thing to put on a linked page for each country, as @michaelsteffen has suggested.
